crates/http/src/config.rs

```diff
@@ -2,16 +2,39 @@ use serde::{Deserialize, Serialize};
 use spin_http_routes::HttpTriggerRouteConfig;
 
 /// Configuration for the HTTP trigger
-#[derive(Clone, Debug, Default, Deserialize, Serialize)]
-#[serde(deny_unknown_fields)]
+#[derive(Clone, Debug, Deserialize, Serialize)]
 pub struct HttpTriggerConfig {
-    /// Component ID to invoke
-    pub component: String,
     /// HTTP route the component will be invoked for
     pub route: HttpTriggerRouteConfig,
-    /// The HTTP executor the component requires
-    #[serde(default)]
-    pub executor: Option<HttpExecutorType>,
+    /// How to handle requests to this route
+    #[serde(flatten)]
+    pub handler: HttpTriggerHandler,
+}
+
+/// Configuration for the HTTP trigger
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case", untagged, deny_unknown_fields)]
```

Comment:
Serde claims that any combination of `deny_unknown_fields` and `flatten` is "not supported" (read: "buggy"). Some alternatives:

- Duplicate `route` in the two variants
- BYO `Deserialize` for `HttpTriggerRouteConfig`
- Go back to a single struct with `component: Option<String>` and `static_response: Option<...>` and validate that exactly one is set later in the trigger